apps/search/js/search.js

```diff
@@ -60,6 +61,20 @@
         });
       };
 
+      // Initialize the port for connecting to the system app to report metrics.
+      navigator.mozApps.getSelf().onsuccess = function() {
+        var app = this.result;
+        app.connect('app-metrics').then(function onConnAccepted(ports) {
+          ports.forEach(function(port) {
+            self._portMetrics = port;
```

Comment:
seems like it would be more effecient to just set `self._portMetrics` directly, i.e

`self._portMetrics = ports[0]`

Is it possible for `ports` to be an empty array? (FWIW, this is my first time seeing app ports in use)